llvm/tools/sycl-post-link/ModuleSplitter.cpp

```diff
@@ -310,10 +339,34 @@ ModuleDesc extractSubModule(const ModuleDesc &MD,
 // points that are specified in ModuleEntryPoints vector.
 ModuleDesc extractCallGraph(const ModuleDesc &MD,
                             EntryPointGroup &&ModuleEntryPoints,
-                            const CallGraph &CG) {
+                            const DependencyGraph &CG,
+                            const std::function<bool(const Function *)>
+                                &IncludeFunctionPredicate = nullptr) {
   SetVector<const GlobalValue *> GVs;
-  collectFunctionsToExtract(GVs, ModuleEntryPoints, CG);
-  collectGlobalVarsToExtract(GVs, MD.getModule());
+  collectFunctionsAndGlobalVariablesToExtract(
+      GVs, MD.getModule(), ModuleEntryPoints, CG, IncludeFunctionPredicate);
+
+  ModuleDesc SplitM = extractSubModule(MD, GVs, std::move(ModuleEntryPoints));
+  SplitM.cleanup();
+
+  return SplitM;
+}
+
+// The function is simlar to 'extracCallGraph', but it produces a coupe of
```

Comment:
Typo and nit
// The function is similar to 'extractCallGraph', but it produces a copy of
// input LLVM IR module M with _all_ ESIMD functions and kernels included,
// regardless of whether or not they are listed in ModuleEntryPoints.

Reply:
Fixed in e067e3cc9c2ae0ef72994e024cc363589e90f96a